static/js/formatters-internal.js

```diff
@@ -311,164 +317,122 @@ export function image(simpleOrComplexImage = {}, desiredSize = '200x', atLeastAs
- *   'https://a.mktgcdn.com/p/ldMLwj1JkN94-2pwh6CjR_OMy4KnexHJCfZhPAZCbi0/196x400.jpg')
  * @param {string} desiredSize The desired size of the image ('<width>x<height>')
  * @param {boolean} atLeastAsLarge Whether the image should be at least as large as the desired
- *                                 size in one dimension or smaller than the desired size in both
- *                                 dimensions.
- * @returns {string} A dynamic url (e.g. 'https://dynl.mktgcdn.com/p/ldMLwj1JkN94-2pwh6CjR_OMy4KnexHJCfZhPAZCbi0/200x1.jpg')
+ *                                 size in both dimensions or at least one dimension.
+ * @param {number?} fullSizeWidth The full size width of the original image if exists
+ * @param {number?} fullSizeHeight The full size height of the original image if exists
+ * @returns {string} A string representing the format options
+ *                   (e.g. 'height=200,width=100,fit=contain')
  */
-function _getUsImageDynamicUrl(imageUrl, desiredSize, atLeastAsLarge) {
-  const [urlWithoutExtension, extension] = _splitStringOnIndex(imageUrl, imageUrl.lastIndexOf('.'));
-  const [urlBeforeDimensions, dimensions] = _splitStringOnIndex(urlWithoutExtension, urlWithoutExtension.lastIndexOf('/') + 1);
-  const fullSizeDims = dimensions.split('x');
-
-  let desiredWidth, desiredHeight;
+function _getImageFormatOptions(desiredSize, atLeastAsLarge, fullSizeWidth, fullSizeHeight) {
   let desiredDims = desiredSize.split('x');
+  const hasDesiredWidth = desiredDims[0] !== '';
+  const hasDesiredHeight = desiredDims[1] !== '';
+  let formatOptions = ['fit=contain'];
 
-  if (desiredDims[0] !== '') {
+  // both dimensions are not provided, return original image
+  if (!hasDesiredWidth && !hasDesiredHeight) {
+    return `/${formatOptions.join(',')}`;
+  }
+
+  let desiredWidth;
+  let desiredHeight;
+  if (hasDesiredWidth) {
     desiredWidth = Number.parseInt(desiredDims[0]);
     if (Number.isNaN(desiredWidth)) {
       throw new Error("Invalid width specified");
     }
-  } else {
-    desiredWidth = atLeastAsLarge ? 1 : Number.parseInt(fullSizeDims[0]);
   }
-
-  if (desiredDims[1] !== '') {
+  if (hasDesiredHeight) {
     desiredHeight = Number.parseInt(desiredDims[1]);
     if (Number.isNaN(desiredHeight)) {
       throw new Error("Invalid height specified");
     }
-  } else {
-    desiredHeight = atLeastAsLarge ? 1 : Number.parseInt(fullSizeDims[1]);
   }
 
-  const urlWithDesiredDims = urlBeforeDimensions + desiredWidth + 'x' + desiredHeight + extension;
+  const originalRatio =
+    (!!fullSizeWidth && !!fullSizeHeight) ? (fullSizeWidth / fullSizeHeight) : undefined;
 
-  return atLeastAsLarge
-    ? _replaceUrlHost(urlWithDesiredDims, 'dynl.mktgcdn.com')
-    : _replaceUrlHost(urlWithDesiredDims, 'dynm.mktgcdn.com');
-}
+  // both dimensions are provided
+  if (hasDesiredWidth && hasDesiredHeight && originalRatio) {
+    const width = atLeastAsLarge
+      ? Math.max(desiredWidth, Math.round(desiredHeight * originalRatio))
```

Comment:
nit: can we condense this logic?
```js
// please feel free to change this naming. didn't spend much time on it
const mathFunc = atLeastAsLarge ? Math.max : Math.min;
const width = mathFunc(desiredWidth, Math.round(...));
const height = mathFunc(desiredHeight, Math.round(...));
```
this way, we can avoid some of the long/repeated ternaries